File: neatformcore/src/main/java/com/nerdstone/neatformcore/views/containers/VerticalRootView.java
package com.nerdstone.neatformcore.views.containers;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import com.nerdstone.neatformcore.domain.model.NFormViewData;
import com.nerdstone.neatformcore.domain.model.form.NFormViewProperty;
import com.nerdstone.neatformcore.domain.view.NFormView;
import com.nerdstone.neatformcore.domain.view.RootView;
import com.nerdstone.neatformcore.views.handlers.ViewDispatcher;

import java.util.List;

import static com.nerdstone.neatformcore.utils.NFormViewUtils.createViews;

public class VerticalRootView extends LinearLayout implements RootView {

    public VerticalRootView(Context context) {
        super(context);
        setupView();
    }

    public VerticalRootView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        setupView();
    }

    @Override
    public RootView initRootView() {
        return this;
    }

    @Override
    public void addChild(NFormView nFormView) {
        this.addView(nFormView.getViewDetails().getView());
    }

    @Override
    public void addChildren(List<NFormViewProperty> viewProperties, ViewDispatcher viewDispatcher) {
        createViews(this, viewProperties, getContext(), viewDispatcher);
    }

    @Override
    public List<NFormViewData> getViewsData() {
        return null;
    }

    @Override
    public void setupView() {
        this.setOrientation(VERTICAL);
    }
}

File: neatformcore/src/main/java/com/nerdstone/neatformcore/views/containers/VerticalRootView.kt
package com.nerdstone.neatformcore.views.containers

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import com.nerdstone.neatformcore.domain.model.NFormViewData
import com.nerdstone.neatformcore.domain.model.NFormViewProperty
import com.nerdstone.neatformcore.domain.view.NFormView
import com.nerdstone.neatformcore.domain.view.RootView
import com.nerdstone.neatformcore.utils.ViewUtils
import com.nerdstone.neatformcore.views.handlers.ViewDispatcher

class VerticalRootView : LinearLayout, RootView {

    init {
        orientation = VERTICAL
    }

    override val viewsData: List<NFormViewData> = listOf()

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun initRootView(): RootView = this

    override fun addChild(nFormView: NFormView) {
        val view = nFormView.viewDetails.view
        val params = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        params.bottomMargin = 20
        view.layoutParams = params
        this.addView(view)
    }

    override fun addChildren(
        viewProperties: List<NFormViewProperty>, viewDispatcher: ViewDispatcher
    ) = ViewUtils.createViews(this, viewProperties, context, viewDispatcher)
}
